Updated k3s version to v1.34.2+k3s1, Calico to v3.31.0 and Cilium to v1.18.5. Fixed some yml's so it all keeps working.

panoptikoe opened the pull request:
# Proposed Changes
<!--- Provide a general summary of your changes -->

- This fixes https://github.com/timothystewart6/k3s-ansible/issues/667
- Changed calico.yml because the CRD's are now split, see https://github.com/projectcalico/calico/blob/release-v3.30/release-notes/v3.30.0-release-notes.md. This is a thing since v3.30.0 and up. Somehow the api-resources for Calico don't seem to be detected, but they do turn up in the cluster itself.
- Tested Cilium, still works fine.
- Modified main.yml in `roles/k3s_server/tasks` so it is compatible with newer versions of k3s.

## Checklist

- [x] Tested locally
- [x] Ran `site.yml` playbook
- [x] Ran `reset.yml` playbook
- [x] Did not add any unnecessary changes
- [x] Ran pre-commit install at least once before committing
- [x] 🚀

panoptikoe committed:
calico.yml modification

Commit:
main.yml modification

Commit:
all.yml modification

capickett commented on inventory/sample/group_vars/all.yml:
```diff
@@ -1,5 +1,5 @@
 ---
-k3s_version: v1.30.2+k3s2
+k3s_version: v1.34.2+k3s2
```
Typo here

```suggestion
k3s_version: v1.34.2+k3s1
```

panoptikoe committed:
Update inventory/sample/group_vars/all.yml

Co-authored-by: Cameron Pickett <pickett.cj@gmail.com>

panoptikoe commented:
Thanks for the suggestions, I've approved them

panoptikoe committed:
Update README with fork details and version changes

Added details about the fork and updated versions for k3s, Calico, and Cilium.

panoptikoe committed:
Update supported OS versions in README

panoptikoe committed:
Update README to link original repo for details

Removed detailed description of the HA k3s Cluster setup and added a link to the original repository's README for more information.

panoptikoe committed:
Add buildk3s.sh script to run Ansible playbook

panoptikoe committed:
Update reset.sh to include inventory and prompt

panoptikoe committed:
Update README with Longhorn feature branch details

Added information about Longhorn deployment feature branch.

panoptikoe committed:
Fix punctuation in README.md

Added a period at the end of the Longhorn branch description.

panoptikoe committed:
Add example deployments section to README

Added a section for example deployments in K3S.

panoptikoe committed:
Revise version details for k3s and Cilium

Updated version information for k3s and Cilium in the README.

panoptikoe committed:
Revise disclaimer for playbook testing environments

Updated disclaimer to clarify testing environments.

panoptikoe committed:
Update README with cloud-init instructions for Rocky and Alma

Added additional instructions for using cloud-init images with Rocky 10 and Alma 10.

panoptikoe committed:
Update k3s version and compatibility notes in README

Updated k3s version to include v1.36.0 and clarified compatibility with various operating systems.

panoptikoe committed:
Update version information for Calico and Cilium

Updated Calico and Cilium versions in the README.

panoptikoe committed:
Add condition to cleanup K3s bootstrap manifests

panoptikoe committed:
Change directory permissions from 0644 to 0755
